amy/autoemails/actions.py

```diff
@@ -1235,3 +1245,283 @@ def get_merge_data(self) -> Optional[Dict]:
+        reminders_to_archive = []
+        for reminder in email_reminders:
+            if reminder.person_id not in people_ids:
+                # if there is an email reminder that is not in people above
+                # archive the email reminder they have logged in
+                reminders_to_archive.append(reminder)
+            elif reminder.number_times_sent == self.REMINDER_LIMIT:
+                # Hit the reminder limit
+                # Archive the person and archive the reminder
+                reminder.person.archive()
+                reminders_to_archive.append(reminder)
+            else:
+                # send email
+                self._update_email_reminder(reminder)
+                self.schedule_rqjob(reminder, triggers)
+                reminder.save()
+        EmailReminder.objects.filter(
+            id__in=[r.id for r in reminders_to_archive]
+        ).update(archived_at=timezone.now())
+
+    def create_new_email_reminders(
+        self,
+        triggers: Iterable[Trigger],
+        email_reminders: Iterable[EmailReminder],
+        people: Iterable[Person],
+    ) -> None:
+        """
+        Creates an email reminder and sends an email
+        for users who have not received a reminder previously.
+        """
+        thirty_days_from_now = timezone.now() + timedelta(days=30)
+        people_already_sent_emails = set([er.person for er in email_reminders])
+        people_new_email_reminders = set(people) - people_already_sent_emails
+        email_reminders = []
+        for person in people_new_email_reminders:
+            email_reminder = EmailReminder(
+                remind_again_date=thirty_days_from_now,
+                person=person,
+                trigger=triggers[0],
+            )
+            self._update_email_reminder(email_reminder)
+            self.schedule_rqjob(email_reminder, triggers)
+            email_reminders.append(email_reminder)
+        EmailReminder.objects.bulk_create(email_reminders)
+
+    def schedule_rqjob(
+        self, reminder: EmailReminder, triggers: Iterable[Trigger]
+    ) -> None:
+        ActionManageMixin.add(
+            action_class=self.EMAIL_ACTION_CLASS,
+            logger=logger,
+            scheduler=scheduler,
+            triggers=triggers,
+            context_objects={
+                "person_email": reminder.person.email,
+                "email_reminder": reminder,
+            },
+            object_=reminder.person,
+        )
+
+    def _update_email_reminder(self, reminder: EmailReminder) -> None:
+        """
+        Updates the date and number of times sent for the email reminder.
+        """
+        # TODO this might have to happen only if the email is successfully sent
+        reminder.remind_again_date = timezone.now() + timedelta(days=30)
+        reminder.number_times_sent += 1
+
+
+class InactiveProfileArchivalWarningRepeatedAction(
+    BaseProfileArchivalWarningRepeatedAction
+):
```

Comment:
Does it have to subclass? `BaseProfileArchivalWarningRepeatedAction` seems very specific for a base class.

Reply:
There is another situation where we send email reminders and archive the person if they do not respond. One is for inactivity (not logging in) and the other is not signing in and consenting to required terms. I'll add that class to this pr.